docs/writing-policies/go/03-policy-settings.md

```diff
@@ -11,160 +11,105 @@ We will do that by adding this code inside of the `settings.go` file:
-we use the `mapset.Set` structure to store the list of denied labels.
+we use the [`mapset.Set`](https://github.com/deckarep/golang-set) type to store
+the list of denied labels.
 
-The `Settings` struct has json attributes, we will use them later when writing
-our unit tests. The unit tests are going to be executed using Go official compiler, hence
-we will be able to leverage the `encoding/json` package.
+The `Settings` struct has no json attributes. That's because none of its
+attributes can be deserialized from JSON:
 
-The `Settings` class is not using the official `regexp.RegExp` object to
-represent regular expressions. That's because the `regexp.RegExp` struct doesn't
-handle serialization and deserialization to JSON.
+  * `regexp.Regexp`: doesn't handle deserialization, not even with the official Go
+    compiler
+  * `mapset.Set`: given it uses `interface`s internally, this cannot be handled by
+    TinyGo due its lack of reflection. Not even `easyjson` can help here
 
-This is the implementation of the `RegularExpression` struct:
+Because of that, we are going to use a different Go type to deserialize the input
+data. We will call this Go type: `BasicSettings`.
 
-```go
-// A wrapper around the standard regexp.Regexp struct
-// that implements marshalling and unmarshalling
-type RegularExpression struct {
-	*regexp.Regexp
-}
 
-// Convenience method to build a regular expression
-func CompileRegularExpression(expr string) (*RegularExpression, error) {
-	nativeRegExp, err := regexp.Compile(expr)
-	if err != nil {
-		return nil, err
-	}
-	return &RegularExpression{nativeRegExp}, nil
-}
+## Defining the `BasicSettings` type
 
-// UnmarshalText satisfies the encoding.TextMarshaler interface,
-// also used by json.Unmarshal.
-func (r *RegularExpression) UnmarshalText(text []byte) error {
-	nativeRegExp, err := regexp.Compile(string(text))
-	if err != nil {
-		return err
-	}
-	r.Regexp = nativeRegExp
-	return nil
-}
+The default policy defines its own `Settings` type inside of the `types.go` file.
+Starting from this file, we then leverage the `easyjson` cli tool to create the
+serialization and deserialization code.
 
-// MarshalText satisfies the encoding.TextMarshaler interface,
-// also used by json.Marshal.
-func (r *RegularExpression) MarshalText() ([]byte, error) {
-	if r.Regexp != nil {
-		return []byte(r.Regexp.String()), nil
-	}
+Open the `types.go` file, remove the `Settings` type and define a new `BasicSettings`
+one:
 
-	return nil, nil
+```go
+type BasicSettings struct {
+	DeniedLabels      []string          `json:"denied_labels"`
+	ConstrainedLabels map[string]string `json:"constrained_labels"`
 }
 ```
+Next, let's regenerate the `types_easyjson.go` file. Usually there's no need to
+do that manually because `make` handles that automatically. However, right now
+we cannot rely on `make` because all the changes we've done so far have broken
```

Comment:
I'm a little confused here as to what we're trying to say? Are we intending to state that in a normal scenario this would work but only while implementing this the first time we need to run the make command? Or is it every single time?
```suggestion
we cannot rely on `make` because all the changes we've done so far have broken
```

Reply:
`make` used to work when we cloned the repository, but since we've started making invasive changes this is temporarily broken. Once we complete all the changes we will be able to use `make` again